### PR TITLE
Porting of SlackDialog class

### PR DESCRIPTION
**Pull request by ceciliaavila**
**Proposed Changes**

* Implement **SlackDialog**'s method signatures.
* Create auxiliar **DialogData** class
* Create auxiliar **DialogElement** class

### CODE REVIEW

File: C#/Botkit/botbuilder-slack-adapter/DialogData.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace botbuilder_slack_adapter
```
Comment (GasparAcevedoZainSouthworks) [inline C#/Botkit/botbuilder-slack-adapter/DialogData.cs]: I think that we can use a namespace like `Microsoft.BotBuilder.SlackAdapter` or `BotBuilder.SlackAdapter`. We can discuss it later

File: C#/Botkit/botbuilder-slack-adapter/SlackDialog.cs [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+        private readonly DialogElement[] elements;
+
+        /// <summary>
+        /// Create a Slack Dialog object for use with [replyWithDialog()](#replyWithDialog)
+        /// </summary>
+        /// <param name="title">The title of the dialog.</param>
+        /// <param name="callback_id">Callback id of dialog.</param>
+        /// <param name="submit_label">Label for the submit button.</param>
+        /// <param name="element">An array of dialog elements.</param>
+        public SlackDialog(string title, string callback_id, string submit_label, List<DialogElement> elements)
```
Comment (GasparAcevedoZainSouthworks) [inline C#/Botkit/botbuilder-slack-adapter/SlackDialog.cs]: Same pattern here

File: C#/Botkit/botbuilder-slack-adapter/DialogElement.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+using System.Text;
+
+namespace botbuilder_slack_adapter
+{
+    public class DialogElement
+    {
+        public string Label { get; set; }
+        public string Name { get; set; }
+        public string Value { get; set; }
+        public Dictionary<string, string> Option_list { get; set; }
```
Comment (GasparAcevedoZainSouthworks) [inline C#/Botkit/botbuilder-slack-adapter/DialogElement.cs]: Same pattern here in `Option_List`. Should be `OptionList`

File: C#/Botkit/botbuilder-slack-adapter/DialogData.cs [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+{
+    public class DialogData
+    {
+        public string Title { get; set; }
```
Comment (VictorGrycuk) [inline C#/Botkit/botbuilder-slack-adapter/DialogData.cs]: Should all these properties be public? In the [original code](https://github.com/howdyai/botkit/blob/master/packages/botbuilder-adapter-slack/src/slack_dialog.ts#L21) `data` was private, and its properties were assigned using the constructor.

File: C#/Botkit/botbuilder-slack-adapter/DialogData.cs [hunk 5]
```diff
@@ -0,0 +1,24 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace botbuilder_slack_adapter
+{
+    public class DialogData
+    {
+        public string Title { get; set; }
+        public string Callback_id { get; set; }
```
[the remark this answers is not in the document]
Comment (ceciliaavila) [reply C#/Botkit/botbuilder-slack-adapter/DialogData.cs]: Done! Thanks!

File: C#/Botkit/botbuilder-slack-adapter/DialogData.cs (hunk 4 above)
Comment (ceciliaavila) [reply C#/Botkit/botbuilder-slack-adapter/DialogData.cs]: We need them to be public so they can be set from SlackDialog class. Thanks!

File: C#/Botkit/botbuilder-slack-adapter/DialogData.cs (hunk 1 above)
Comment (GasparAcevedoZainSouthworks) [reply C#/Botkit/botbuilder-slack-adapter/DialogData.cs]: NOTE: we will update this in another PR.

**Pull request merged**